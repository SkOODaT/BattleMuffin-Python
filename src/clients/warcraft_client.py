import requests
from authlib.integrations.requests_client import OAuth2Session
from uplink import Consumer, Query, get, params, returns

from config.client_config import ClientConfiguration
from config.region_config import Locale, Region


class WarcraftClient(Consumer):
    def __init__(self, client_id: str, client_secret: str, region: Region = Region.us, locale: Locale = Locale.en_US):
        self.client_config = ClientConfiguration(client_id, client_secret, region, locale)

        oath_response = requests.get(f"{self.client_config.oauth_host}/oauth/.well-known/openid-configuration").json()
        token_endpoint = oath_response['token_endpoint']
        session = OAuth2Session(client_id, client_secret, scope="openid", token_endpoint=token_endpoint, grant_type="client_credentials")
        session.fetch_token()
        super().__init__(base_url=self.client_config.host, client=session)
        self._inject(Query("locale").with_value(self.client_config.locale.name))

    @returns.json
    @params({"namespace": "static-us"})
    @get("/data/wow/achievement-category/index")
    def get_achievement_category_index(self):
        """Returns an index of achievement categories."""
