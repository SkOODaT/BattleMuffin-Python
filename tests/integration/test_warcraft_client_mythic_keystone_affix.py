import os

from battlemuffin.clients.warcraft_client import WarcraftClient
from battlemuffin.config.region_config import Locale, Region


def test_get_mythic_keystone_affixes_index(snapshot):
    client = WarcraftClient(
        os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"), Region.us, Locale.en_US
    )
    response = client.get_mythic_keystone_affixes_index()
    assert response == snapshot


def test_get_mythic_keystone_affix(snapshot):
    client = WarcraftClient(
        os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"), Region.us, Locale.en_US
    )
    response = client.get_mythic_keystone_affix(1)
    assert response == snapshot


def test_get_mythic_keystone_affix_media(snapshot):
    client = WarcraftClient(
        os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"), Region.us, Locale.en_US
    )
    response = client.get_mythic_keystone_affix_media(1)
    assert response == snapshot
