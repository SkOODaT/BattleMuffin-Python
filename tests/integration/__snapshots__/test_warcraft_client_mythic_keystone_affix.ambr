# name: test_get_mythic_keystone_affix
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/1?namespace=static-8.3.0_32861-us',
      },
    },
    'description': "Healing in excess of a target's maximum health is instead converted to a heal absorption effect.",
    'id': 1,
    'media': <class 'dict'> {
      'id': 1,
      'key': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/media/keystone-affix/1?namespace=static-8.3.0_32861-us',
      },
    },
    'name': 'Overflowing',
  }
---
# name: test_get_mythic_keystone_affix_media
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/media/keystone-affix/1?namespace=static-8.3.0_32861-us',
      },
    },
    'assets': <class 'list'> [
      <class 'dict'> {
        'key': 'icon',
        'value': 'https://render-us.worldofwarcraft.com/icons/56/inv_misc_volatilewater.jpg',
      },
    ],
    'id': 1,
  }
---
# name: test_get_mythic_keystone_affixes_index
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/?namespace=static-8.3.0_32861-us',
      },
    },
    'affixes': <class 'list'> [
      <class 'dict'> {
        'id': 1,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/1?namespace=static-8.3.0_32861-us',
        },
        'name': 'Overflowing',
      },
      <class 'dict'> {
        'id': 2,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/2?namespace=static-8.3.0_32861-us',
        },
        'name': 'Skittish',
      },
      <class 'dict'> {
        'id': 3,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/3?namespace=static-8.3.0_32861-us',
        },
        'name': 'Volcanic',
      },
      <class 'dict'> {
        'id': 4,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/4?namespace=static-8.3.0_32861-us',
        },
        'name': 'Necrotic',
      },
      <class 'dict'> {
        'id': 5,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/5?namespace=static-8.3.0_32861-us',
        },
        'name': 'Teeming',
      },
      <class 'dict'> {
        'id': 6,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/6?namespace=static-8.3.0_32861-us',
        },
        'name': 'Raging',
      },
      <class 'dict'> {
        'id': 7,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/7?namespace=static-8.3.0_32861-us',
        },
        'name': 'Bolstering',
      },
      <class 'dict'> {
        'id': 8,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/8?namespace=static-8.3.0_32861-us',
        },
        'name': 'Sanguine',
      },
      <class 'dict'> {
        'id': 9,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/9?namespace=static-8.3.0_32861-us',
        },
        'name': 'Tyrannical',
      },
      <class 'dict'> {
        'id': 10,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/10?namespace=static-8.3.0_32861-us',
        },
        'name': 'Fortified',
      },
      <class 'dict'> {
        'id': 11,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/11?namespace=static-8.3.0_32861-us',
        },
        'name': 'Bursting',
      },
      <class 'dict'> {
        'id': 12,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/12?namespace=static-8.3.0_32861-us',
        },
        'name': 'Grievous',
      },
      <class 'dict'> {
        'id': 13,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/13?namespace=static-8.3.0_32861-us',
        },
        'name': 'Explosive',
      },
      <class 'dict'> {
        'id': 14,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/14?namespace=static-8.3.0_32861-us',
        },
        'name': 'Quaking',
      },
      <class 'dict'> {
        'id': 15,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/15?namespace=static-8.3.0_32861-us',
        },
        'name': 'Relentless',
      },
      <class 'dict'> {
        'id': 16,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/16?namespace=static-8.3.0_32861-us',
        },
        'name': 'Infested',
      },
      <class 'dict'> {
        'id': 117,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/117?namespace=static-8.3.0_32861-us',
        },
        'name': 'Reaping',
      },
      <class 'dict'> {
        'id': 119,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/119?namespace=static-8.3.0_32861-us',
        },
        'name': 'Beguiling',
      },
      <class 'dict'> {
        'id': 120,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/keystone-affix/120?namespace=static-8.3.0_32861-us',
        },
        'name': 'Awakened',
      },
    ],
  }
---
